src/qml/BookmarkRenderer.qml

```diff
@@ -124,24 +124,37 @@ Item {
           }
         }
 
-        MouseArea {
-          anchors.fill: bookmark
-          onClicked: {
-            displayToast(qsTr('Bookmark: %1').arg(bookmarkRenderer.bookmarkName));
-          }
-          onDoubleClicked: {
-            bookmarkModel.setExtentFromBookmark(bookmarkModel.index(bookmarkRenderer.bookmarkIndex, 0));
-          }
-          onPressAndHold: {
-            if (bookmarkRenderer.bookmarkUser) {
-              bookmarkProperties.bookmarkId = bookmarkRenderer.bookmarkId;
-              bookmarkProperties.bookmarkName = bookmarkRenderer.bookmarkName;
-              bookmarkProperties.bookmarkGroup = bookmarkRenderer.bookmarkGroup;
-              bookmarkProperties.open();
-            } else {
-              displayToast(qsTr('Project bookmarks cannot be edited'));
-            }
-          }
+        Component.onCompleted: {
+          // Register bookmark handler
+          mapCanvasPointHandler.registerHandler("bookmark_" + bookmarkRenderer.bookmarkId, (point, type, interactionType) => {
```

Comment:
I had envisioned one function sitting at the root of the bookmark renderer, and loop through the repeater to check for overlap. But this also works, and I think might even be better so good thinking :)